modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/VirtualFlowTest.java

```diff
@@ -1207,17 +1226,177 @@ public void testSheetChildrenRemainsConstant() {
+        // contains a Circle whith a radius that varies between cells.
+        VirtualFlowShim<IndexedCell> circleFlow;
+        circleFlow = new VirtualFlowShim();
+
+        circleFlow.setVertical(true);
+        circleFlow.setCellFactory(p -> new GraphicalCellStub() {
+            @Override
+            protected double computeMinWidth(double height) {
+                return computePrefWidth(height);
+            }
+
+            @Override
+            protected double computeMaxWidth(double height) {
+                return computePrefWidth(height);
+            }
+
+            @Override
+            protected double computePrefWidth(double height) {
+                return super.computePrefWidth(height);
+            }
+
+            @Override
+            protected double computeMinHeight(double width) {
+                return computePrefHeight(width);
+            }
+
+            @Override
+            protected double computeMaxHeight(double width) {
+                return computePrefHeight(width);
+            }
+
+        });
+        circleFlow.setCellCount(7);
+        circleFlow.resize(300, 300);
+        circleFlow.layout();
+        circleFlow.layout();
+        return circleFlow;
+    }
+
+    // when moving the flow in one direction, the position of the flow
+    // should not increase in the opposite direction
+    @Test
+    public void testReverseOrder() {
+        double orig = flow.getPosition();
+        flow.scrollPixels(10);
+        double pos = flow.getPosition();
+        assertFalse("Moving in positive direction should not decrease position", pos < orig);
+        flow.scrollPixels(-50);
+        double neg = flow.getPosition();
+        assertFalse("Moving in negative direction should not decrease position", neg > pos);
+    }
+
+    @Test
+    public void testReverseOrderForCircleFlow() {
+        VirtualFlowShim vf = createCircleFlow();
+        double orig = vf.getPosition();
+        vf.scrollPixels(10);
+        double pos = vf.getPosition();
+        assertFalse("Moving in positive direction should not decrease position", pos < orig);
+        vf.scrollPixels(-50);
+        double neg = vf.getPosition();
+        assertFalse("Moving in negative direction should not decrease position", neg > pos);
+    }
+
+    @Test
+    public void testGradualMoveForCircleFlow() {
+        VirtualFlowShim vf = createCircleFlow();
+        vf.resize(600,400);
+        ScrollBar sb = vf.shim_getVbar();
+        double s0 = sb.getLayoutY();
+        double s1 = s0;
+        double position = vf.getPosition();
+        double newPosition = 0d;
+        double delta = 0;
+        double newDelta = 0;
+        vf.layout();
+        for (int i = 0; i < 50; i++) {
+            vf.scrollPixels(10);
+            vf.layout();
+            newPosition = vf.getPosition();
+            s1 = sb.getLayoutY();
+            newDelta = newPosition - position;
+            System.err.println("s0 = "+s0+", s1 = "+s1);
```

Comment:
I suggest to comment out all System.err.println() calls from this loop.
It might be useful while debugging individual failing test, but keeping them on for every test run will simply flood the log.
Similar logs were fixed for stdout earlier. Please refer - https://bugs.openjdk.java.net/browse/JDK-8255497

Reply:
System.err.println()s are removed now.